src/CoreWCF.Http/src/CoreWCF/Channels/ServiceModelHttpMiddleware.cs

```diff
@@ -171,6 +172,10 @@ private RequestDelegate BuildBranch()
                         {
                             wcfApp.UseWebSockets(requestHandler.WebSocketOptions);
                         }
+                        if (SecurityUtils.ShouldUseAuthentication(dispatcher.Binding))
+                        {
```

Comment:
Which should go before the WebSocketOptions so that WebSockets is authenticated too.

Reply:
Take a look at RequestDelegateHandler.BuildHandler, it already extracts the HttpTransportBindingElement and stores the AuthenticationScheme in HttpTransportSettings (in the field _httpSettings) so I think ShouldUseAuthentication would be better to exist on RequestDelegateHandler as it already has all the information available.